integrations/jira/jira/client.py

```diff
@@ -103,107 +167,126 @@ async def create_events_webhook(self, app_host: str) -> None:
-            )
+        cursor = None
+        while True:
+            params: Dict[str, Any] = {}
+            if cursor:
+                params["cursor"] = cursor
 
-        params["maxResults"] = PAGE_SIZE
-        while params["startAt"] < total_users:
-            logger.info(f"Current query position: {params['startAt']}/{total_users}")
+            teams_data = await self._send_api_request("GET", base_url, params=params)
 
-            user_response_list = await self._get_users_data(params)
+            if not teams_data["entities"]:
+                break
 
-            if not user_response_list:
-                logger.warning(f"No users found at {params['startAt']}")
+            logger.info(f"Retrieved {len(teams_data['entities'])} teams")
+            yield teams_data["entities"]
+
+            cursor = teams_data.get("cursor")
+            if not cursor:
                 break
 
-            logger.info(
-                f"Retrieved users: {len(user_response_list)} "
-                f"(Position: {params['startAt']}/{total_users})"
-            )
+    async def get_paginated_team_members(
+        self, team_id: str, page_size: int = 40
+    ) -> AsyncGenerator[List[Dict[str, Any]], None]:
+        logger.info(f"Getting members for team {team_id}")
+        org_id = ocean.integration_config["atlassian_organisation_id"]
+        url = f"{self.jira_url}/gateway/api/public/teams/v1/org/{org_id}/teams/{team_id}/members"
 
-            yield user_response_list
-            params["startAt"] += PAGE_SIZE
+        async for members in self._get_cursor_paginated_data(
+            url,
+            method="POST",
+            extract_key="results",
+            page_size=page_size,
+            initial_params={"first": page_size},
+            cursor_param="after",
+        ):
+            logger.info(f"Retrieved {len(members)} members for team {team_id}")
+            yield members
 
-    async def get_single_user(self, account_id: str) -> dict[str, Any]:
-        user_response = await self.client.get(
-            f"{self.api_url}/user", params={"accountId": account_id}
-        )
-        user_response.raise_for_status()
-        return user_response.json()
+    async def get_user_team_mapping(self) -> Dict[str, str]:
+
+        user_team_mapping = {}
+
+        teams = []
+        async for team_batch in self.get_paginated_teams():
+            teams.extend(team_batch)
+
+        logger.info(f"Processing {len(teams)} teams for user mapping")
+
+        for team in teams:
+            team_id = team["teamId"]
+            async for members in self.get_paginated_team_members(team_id):
+                for member in members:
+                    account_id = member["accountId"]
+                    if account_id not in user_team_mapping:
+                        user_team_mapping[account_id] = team_id
+
+        logger.info(f"Created mapping for {len(user_team_mapping)} users")
+        return user_team_mapping
```

Comment:
can you help me understand what this method is doing?

Reply:
It Gets all teams in the organization using, then Collects all teams into a single list
For each team: It Gets all members of that team using pagination
For each member in the team: Adds the user-team mapping , to get something like this 
{
    "user123": "team456",
    "user789": "team456",
    "user234": "team789"
}

This isnt ideal as it only maps the the user to the first team it finds for it, Im working on a fix for this